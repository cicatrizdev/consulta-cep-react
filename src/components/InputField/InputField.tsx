import React from 'react'
import styled from 'styled-components'
import InputMask from 'react-input-mask'
import { omit } from 'lodash'

interface IInputField {
  id: string
  onChange: any
  value: string
}

const Wrapper = styled(InputMask)`
  border: 1px solid darkgray;
  border-radius: 5px;
  box-shadow: 0;
  font-size: 1rem;
  padding: 5px 10px;
`

const InputField = (props: IInputField) => {
  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    props.onChange(e.target.value)
  }

  return (
    <Wrapper
      {...omit(props, 'onChange')}
      mask='99999-999'
      onChange={onChange}
      data-testid='search-field'
      placeholder='01234-567'
      required
    />
  )
}

export default InputField;
