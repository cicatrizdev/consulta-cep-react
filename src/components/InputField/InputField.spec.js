import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import InputField from './InputField'

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<InputField />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});