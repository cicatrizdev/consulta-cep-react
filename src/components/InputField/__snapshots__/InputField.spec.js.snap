// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<input
  className="sc-bdVaJa kThWmV"
  data-testid="search-field"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onPaste={[Function]}
  placeholder="01234-567"
  required={true}
/>
`;
